app/models/vacancy.rb

```diff
@@ -37,15 +37,59 @@ class Vacancy < ApplicationRecord
+  # consumes an Algolia operation to try and look it up if it appears in canonical list of records returned by the
+  # model.
+  #
+  # To illustrate: if you run the unmodified `Vacancy.reindex!` on a recent (2020-06-18) production dataset you will
+  # consume more than 30,000 operations on the Alogolia app. This occurs because it looks up each of the 30,000+
+  # expired/unpublished records before it applies the `:listed?` filter. It only indexes about 470 records. I am not
+  # 100% certain, but it seems this is done so it can remove records that should not be in the index according to the
+  # filter.
+  #
+  # If, however, you run `Vacancy.live.reindex!`, which scopes the list to only the "published" records, it only
+  # consumes slightly more operation than there are indexable records.
+  def self.reindex!
+    live.algolia_reindex!
+  end
+
+  def self.reindex
+    live.algolia_reindex
+  end
+
+  # This is intended as a one-shot method used in conjuntion with `algolia_index...if: :listed?` to clear ununsed
+  # records from the production db. It should be deleted after it is successfully run.
+  def self.full_reindex!
+    algolia_reindex!
+  end
+
+  # This is the main method you should use most of the time when bulk-adding new records to the algolia index. It will
+  # not use any additional operations checking records that have been indexed once. NOTE: if a record has been indexed
+  # already and it is updated with new or additional information, the `auto_index: true` will do the work of keeping the
+  # changes in sync with the algolia index. This method is solely for preventing us paying for unnecessary usage when
+  # adding records that have become `live` since the last time it was run.
+  def self.update_index!
+    unindexed.algolia_reindex!
+    # rubocop:disable Rails/SkipsModelValidations
+    unindexed.update_all(initially_indexed: true)
+    # rubocop:enable Rails/SkipsModelValidations
+  end
+
+  # I'm excluding expires_on from the where clause as expiry_time seems to be exactly tracking it-as expected.
+  # TODO: remove expires_on completely from the attributes and only use expiry_time. Ticket to follow.
+  def self.remove_vacancies_that_expired_yesterday!
+    expired_records = where('expiry_time BETWEEN ? AND ?', Time.zone.yesterday.midnight, Time.zone.today.midnight)
+    index.delete_objects(expired_records.map(&:id)) if expired_records.present?
+  end
 
   # rubocop:disable Metrics/BlockLength
   # rubocop:disable Metrics/LineLength
-  # There must be a better way to pass these settings to the block, but everything seems to break
-  algoliasearch index_name: Rails.env.test? ? "Vacancy_test#{ENV.fetch('GITHUB_RUN_ID', '')}" : 'Vacancy', auto_index: Rails.env.production?, auto_remove: Rails.env.production?, synchronous: Rails.env.test?, disable_indexing: !(Rails.env.production? || Rails.env.test?) do
+  algoliasearch auto_index: true, auto_remove: true, if: :listed? do
```

Comment:
I think `auto_index` and `auto_remove` are true by default

Reply:
Noted.  Doesn't hurt to make them explicit :)